File: Cpp/1248_Count_Number_of_Nice_Subarrays.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        int result = 0;
        int odd = 0;
        unordered_map<int, int> count;
        count[0] = 1;
        for(int i=0; i<nums.size(); ++i){
            odd += nums[i] % 2;
            if(count.find(odd - k) != count.end()){
                result += count[odd - k];
            }
            ++count[odd];
        }
        
        return result;
    }
};

File: Kotlin/1248_Count_Number_of_Nice_Subarrays.kt
class Solution {
    fun numberOfSubarrays(nums: IntArray, k: Int): Int {
        var result = 0
        var odd = 0
        val count = HashMap<Int,Int>()
        count[0] = 1
        
        nums.forEach{
            odd += it % 2
            result += count.get(odd - k) ?: 0
            count.merge(odd, 1, Int::plus)
        }
        
        return result
    }
}